primary/src/grpc_server.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::time::Duration;
+
+use crate::block_waiter::{BlockError, BlocksError, GetBlockResponse, GetBlocksResponse};
+use crate::BlockCommand;
+use tokio::sync::mpsc::Sender;
+use tokio::sync::oneshot;
+use tokio::time::sleep;
+use tonic::{transport::Server, Request, Response, Status};
+use tracing::error;
+use types::{
+    BatchMessageProto, CertificateDigest, CollectionRetrievalResult, GetCollectionsRequest,
+    GetCollectionsResponse, Validator, ValidatorServer,
+};
+
+#[derive(Debug)]
+pub struct Narwhal {
+    tx_get_block_commands: Sender<BlockCommand>,
+}
+
+impl Narwhal {
+    fn new(tx_get_block_commands: Sender<BlockCommand>) -> Self {
+        Self {
+            tx_get_block_commands,
+        }
+    }
+}
+
+#[tonic::async_trait]
+impl Validator for Narwhal {
```

Comment:
Assuming that we move the gRPC implementation under its own module, then it might worth having each service (`Validator`, `Proposer` etc) on their own files as well

Reply:
@akichidis what do you mean by `own module`? This implementation is already inside its own module named `primary::grpc_server`. Are you instead asking if this should go inside its own crate?

Reply:
Apologies should have made my self more clear. I mean organise in a folder structure like:
```
......
|-- grpc_server
|.    |- mod.rs
|.    |- validator.rs
|.    |- proposer.rs    
|
```
see the `block_synchronizer`